backend/services/battles.py

```diff
@@ -39,9 +39,13 @@ def set_battle_winner(battle):
     creator = Team.objects.get(battle=battle, trainer=battle.creator.id)
     opponent = Team.objects.get(battle=battle, trainer=battle.opponent.id)
 
-    # get the pokemons names from each team
-    creator_team = creator.pokemons.all()
-    opponent_team = opponent.pokemons.all()
+    # get pokemon in the right order
+    creator_pokemon_list = PokemonTeam.objects.filter(team=creator)
+    opponent_pokemon_list = PokemonTeam.objects.filter(team=opponent)
```

Comment:
How the order is ensured here if you're not calling `order_by`?

Reply:
Also, I think this query can be optimized with:
`PokemonTeam.objects.filter(team__battle=battle, team__trainer=opponent).select_related('pokemon')`

Same for the query above.

Reply:
I've done a few tests and it was behaving in the right order, but I changed to call it just to make sure!! :D